core/templates/dev/head/galleries/gallery.html

```diff
@@ -156,23 +156,41 @@
             <!-- TODO(sll): Add card at end of the row that, when clicked, does showFullGalleryGroup(galleryGroup)" -->
-            <h2 class="oppia-gallery-group-header">
-              <a ng-click="showFullGalleryGroup(galleryGroup)">
-                <[galleryGroup.header]>
-              </a>
-            </h2>
-            <exploration-summary-tile ng-repeat="tile in galleryGroup.activity_summary_dicts"
-                                      exploration-id="tile.id"
-                                      exploration-title="tile.title"
-                                      last-updated-msec="tile.last_updated_msec"
-                                      objective="tile.objective"
-                                      category="tile.category"
-                                      ratings="tile.ratings"
-                                      thumbnail-icon-url="tile.thumbnail_icon_url"
-                                      thumbnail-bg-color="tile.thumbnail_bg_color">
-            </exploration-summary-tile>
+            <div>
+              <h2 class="oppia-gallery-group-header">
+                <a ng-click="showFullGalleryGroup(galleryGroup)">
+                  <[galleryGroup.header]>
+                </a>
+              </h2>
+            </div>
+            <md-button class="md-no-ink oppia-gallery-scroll-btn" ng-click="scroll($index, true)"
+                       ng-hide='rowIndices[$index] == 0'>
+              <i class="material-icons">keyboard_arrow_left</i>
+            </md-button>
+            <div class="oppia-gallery-scroll-btn" ng-show="rowIndices[$index] == 0"></div>
+            <div class="oppia-gallery-tiles-carousel" ng-swipe-right="scroll($index, true)" ng-swipe-left="scroll($index, false)">
+              <div class="oppia-gallery-tiles-carousel-overlay-left"></div>
+              <div class="oppia-gallery-tiles">
+                <exploration-summary-tile ng-repeat="tile in galleryGroup.activity_summary_dicts"
+                                          exploration-id="tile.id"
+                                          exploration-title="tile.title"
+                                          last-updated-msec="tile.last_updated_msec"
+                                          objective="tile.objective"
+                                          category="tile.category"
+                                          ratings="tile.ratings"
+                                          thumbnail-icon-url="tile.thumbnail_icon_url"
+                                          thumbnail-bg-color="tile.thumbnail_bg_color">
+                </exploration-summary-tile>
+              </div>
+              <div class="oppia-gallery-tiles-carousel-overlay-right"></div>
+            </div>
+            <md-button class="md-no-ink oppia-gallery-scroll-btn" ng-click="scroll($index, false)"
+                       ng-hide="(galleryGroup.activity_summary_dicts.length - tileDisplayCount) <= rowIndices[$index]">
```

Comment:
Can an ng-if be used here, instead of ng-hide? Otherwise the cursor still changes as you move over the area.

Ditto for the other button.